system_tests/test_vsphere_local_blueprints/local_linux_test.py

```diff
@@ -353,6 +524,208 @@ def test_distributed_network(self):
+
+        self.logger.info('Attempting to deploy with invalid network name')
+
+        inputs = copy(self.ext_inputs)
+        inputs['test_network_distributed'] = False
+        inputs['test_network_name'] = 'notarealnetworkdonotuse'
+
+        self.invalid_network_env = local.init_env(
+            blueprint,
+            inputs=inputs,
+            name=self._testMethodName,
+            ignored_modules=cli_constants.IGNORED_LOCAL_WORKFLOW_MODULES)
+
+        try:
+            self.invalid_network_env.execute(
+                'install',
+                task_retries=50,
+                task_retry_interval=3,
+            )
+            self.invalid_network_env.execute(
+                'uninstall',
+                task_retries=50,
+                task_retry_interval=3,
+            )
+            raise AssertionError(
+                'Deploying with an invalid network was expected to fail, but '
+                'succeeded. Network name was {}'.format(
+                    inputs['test_network_name'],
+                )
+            )
+        except RuntimeError as err:
+            # Ensure the error message has pertinent information
+            assert inputs['test_network_name'] in err.message
+            assert 'not present' in err.message
+            assert 'Available networks are:' in err.message
+
+    def test_incorrect_switch_distributed_true(self):
+        blueprint = os.path.join(
+            self.blueprints_path,
+            'network-fail-blueprint.yaml'
+        )
+
+        if self.env.install_plugins:
+            self.logger.info('installing required plugins')
+            self.cfy.install_plugins_locally(
+                blueprint_path=blueprint)
+
+        self.logger.info(
+            'Attempting to deploy on standard network with '
+            'switch_distributed set to true'
+        )
+
+        inputs = copy(self.ext_inputs)
+        inputs['test_network_distributed'] = True
+        inputs['test_network_name'] = (
+            self.env.cloudify_config['existing_standard_network']
+        )
+
+        self.incorrect_distributed_true_env = local.init_env(
+            blueprint,
+            inputs=inputs,
+            name=self._testMethodName,
+            ignored_modules=cli_constants.IGNORED_LOCAL_WORKFLOW_MODULES)
+
+        try:
+            self.incorrect_distributed_true_env.execute(
+                'install',
+                task_retries=50,
+                task_retry_interval=3,
+            )
+            self.incorrect_distributed_true_env.execute(
+                'uninstall',
+                task_retries=50,
+                task_retry_interval=3,
+            )
+            raise AssertionError(
+                'Deploying with an invalid distributed network was expected '
+                'to fail, but succeeded. Network name was {}'.format(
+                    inputs['test_network_name'],
+                )
+            )
+        except RuntimeError as err:
```

Comment:
I think I was not using that because we're using the non-unittest thing in some places and this is parented off cosmo_tester, so if that were changed to use the non-unittest one then this is somewhere it'd break really badly.

I'm inclined to leave it like this until we make a firm decision on how the system tests are being approached rather than changing it now and then finding that we've settled on something that means it needs changing back or differently in a month or three.